### PR TITLE
Fix for the CI

### PR DESCRIPTION
**Pull request by JovanVeljanoski**
Yet another PR to fix the CI. Currently xgboost is giving trouble on mac.

Addresses #748

### CODE REVIEW

File: .github/workflows/pythonpackage.yml [hunk 1]
```diff
@@ -46,7 +46,7 @@ jobs:
     - name: Create env
       # if: steps.cache-conda-env.outputs.cache-hit != 'true'
       run: |
-        conda create -q -n vaex -c conda-forge python=${{ matrix.python-version }} numpy scipy xarray pyqt matplotlib pyopengl h5py astropy tornado cython pandas pytest numba pyarrow graphviz python-graphviz pcre catboost plotly notebook scikit-learn lightgbm py-xgboost
```
Comment (xhochy) [inline .github/workflows/pythonpackage.yml:49]: You should be able to revert that again, everything should be fixed now.

File: .travis.yml [hunk 2]
```diff
@@ -36,12 +36,16 @@ before_install:
...
   - conda info -a
   # future: kapteyn should only be needed for astro package testing
-  - conda create -q -n test-environment -c conda-forge python=$PYTHON_VERSION setuptools==42.0.2 certifi=2019.6.16 pip numpy scipy xarray pyqt matplotlib pyopengl h5py astropy tornado cython pandas pytest pytest-asyncio numba pyarrow graphviz python-graphviz pcre catboost libcxx=9.0.0 notebook scikit-learn lightgbm py-xgboost # $OPTIONAL_PACKAGES
+  - conda create -q -n test-environment -c conda-forge python=$PYTHON_VERSION setuptools==42.0.2 certifi=2019.6.16 pip numpy scipy xarray pyqt matplotlib pyopengl h5py astropy tornado cython pandas pytest pytest-asyncio numba graphviz python-graphviz pcre catboost libcxx=9.0.0 notebook scikit-learn lightgbm py-xgboost # $OPTIONAL_PACKAGES
   - which pip
   - source activate test-environment
   - if [[ "$TRAVIS_OS_NAME" == "linux" ]]; then
       conda install gxx_linux-64 -y;
     fi
+  # Install OpenMP on osx with brew
```
Comment (xhochy) [inline .travis.yml:45]: You can get that via `conda` as well, it is named `llvm-openmp`

Comment (maartenbreddels) [reply .travis.yml:45]: Cool, that is much cleaner, how should we have known this btw? We tried 'omp' I believe, but was kind of a mystery.

Comment (xhochy) [reply .travis.yml]: Checking in the jungle of the conda-forge documentation: https://conda-forge.org/docs/maintainer/knowledge_base.html#openmp

Also it seems like you are only using the C-F compiler on Linux, might be that `llvm-openmp` is not compatible with the system compilers. Generally, you should use conda-forge packages only with the conda-forge provided compiler (you can get them either via the `gxx_linux-64` names or use the convenience package `compilers` which will get the correct ones platform-independently.

Comment (JovanVeljanoski) [reply .travis.yml]: Thanks, we are trying of openmp via conda-forge now as you suggested. After that I will try pyarrow via conda-forge as well.

Comment (maartenbreddels) [reply .travis.yml]: So we can replace `conda install gxx_linux-64 -y;` by `conda install -c conda-forge compilers -y;` for all OS'es?

Comment (xhochy) [reply .travis.yml]: Yes, this will have the disadvantage that it will also install a fortran compiler but I think this is worth the comfort.

Comment (JovanVeljanoski) [reply .travis.yml]: Cool, here we go again :)

Comment (maartenbreddels) [reply .travis.yml]: Not sure why that led to a compilation issue with annoy :|

**Comment by maartenbreddels** [comment]
Both CI issues are unrelated, I suggest we merge this first.

**Pull request merged**